src/store/modules/stage.ts

```diff
@@ -0,0 +1,189 @@
+
+const StageModule: Module<StageState, {}> = {
+  namespaced: true,
+  state () {
+    return {
+      stageZoomMax: 300,
+      stageZoomMin: 10,
+      stageZoomStep: 10,
+      stageZoom: 100,
+      stageConfig: {
+        width: 0,
+        height: 0,
+        initialWidth: 0,
+        initialHeight: 0,
+        scale: {
+          x: 1,
+          y: 1
+        }
+      }
+    }
+  },
+  getters: {
+    [StageGetters.STAGE_ZOOM]: state => state.stageZoom,
+    [StageGetters.STAGE_CONFIG]: state => state.stageConfig,
+    [StageGetters.STAGE_ZOOM_STEP]: state => state.stageZoomStep
+  },
+  mutations: {
+    [StageMutations.SET_ZOOM] (state: StageState, newValue: number) {
+      if (newValue < state.stageZoomMin) {
+        state.stageZoom = state.stageZoomMin
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      } else if (newValue > state.stageZoomMax) {
+        state.stageZoom = state.stageZoomMax
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      } else {
+        state.stageZoom = newValue
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      }
+    },
+    [StageMutations.SET_ZOOM_IN] (state: StageState) {
+      const newValue = state.stageZoom + state.stageZoomStep
+      if (newValue < state.stageZoomMin) {
+        state.stageZoom = state.stageZoomMin
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      } else if (newValue > state.stageZoomMax) {
+        state.stageZoom = state.stageZoomMax
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      } else {
+        state.stageZoom = newValue
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      }
+    },
+    [StageMutations.SET_ZOOM_OUT] (state: StageState) {
+      const newValue = state.stageZoom - state.stageZoomStep
+      if (newValue < state.stageZoomMin) {
+        state.stageZoom = state.stageZoomMin
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      } else if (newValue > state.stageZoomMax) {
+        state.stageZoom = state.stageZoomMax
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      } else {
+        state.stageZoom = newValue
+        state.stageConfig.scale = {
+          x: (state.stageZoom / 100),
+          y: (state.stageZoom / 100)
+        }
+      }
+    },
+    [StageMutations.SET_CONFIG_SCALE] (state: StageState, newValue: number) {
+      if (newValue < (state.stageZoomMin) / 100) {
```

Comment:
Gets a little repetitive here. Does anyone know how to improve this? I dont know whats good and bad practice inside a module so would like some help here. Works for now